js/src/Ice/ConnectionI.js

```diff
@@ -1338,12 +1319,13 @@ export class ConnectionI {
             switch (messageType) {
                 case Protocol.closeConnectionMsg: {
                     TraceUtil.traceRecv(info.stream, this._logger, this._traceLevels);
+                    // We transition directly to StateClosed, not StateClosingPending.
```

Comment:
In other language mappings, we transition to StateClosingHolding then (usually) immediately to StateClosed. See for example:

https://github.com/zeroc-ice/ice/blob/3fd9cdc81a2d3adbd666fb941777050cf1c8a102/csharp/src/Ice/ConnectionI.cs#L2301

There is no Transceiver.closing in JS. When comparing with other languages, look at the behavior of the TcpTransceiver.closing:

https://github.com/zeroc-ice/ice/blob/3fd9cdc81a2d3adbd666fb941777050cf1c8a102/csharp/src/Ice/Internal/TcpTransceiver.cs#L25

Note that we stop sending heartbeats once we receive CloseConnection from the peer, and StateClosingPending and StateClosed are equivalent in this respect.